apricot/local/service.go

```diff
@@ -52,9 +54,96 @@ type Service struct {
 	src cfgbackend.Source
 }
 
+type machineInfo struct {
+	name string
+}
+
+type clusterInfo struct {
+	FLPs []machineInfo
+}
+
+// Just a skeleton of a function for getting FLP names if relevant
+/*func queryMachines(cluster *clusterInfo) error {
+	for rows.Next() {
+		machine := machineInfo{}
+		err = rows.Scan(&machineInfo.name)
+		if err != nil {
+			return err
+		}
+		cluster.FLPs = append(cluster.FLPs, machine)
+	}
+	err = rows.Err()
+	if err != nil {
+		return err
+	}
+	return nil
+}*/
+
+func ApiGetClusterInformation(w http.ResponseWriter, r *http.Request) {
+	queryParam := mux.Vars(r)
+    format := ""
+    var err error
+	format, err = queryParam["format"]
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "Error, format is not readable.")
+		return
+	}
+	switch format {
+	case "json":
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		//TODO: write JSON
+		/* where to get info? depending on answer, could proceed like:
```

Comment:
See my comment concerning the constructor below. Now that you have an instance (or otherwise a pointer to the Apricot service instance) your function can become `func (ts *TrivialService) ApiGetClusterInformation(w, r)`, and then in the function body you can use `ts.svc` to acquire and then format the inventory.